internal/test/scorecard/common_utils.go

```diff
@@ -453,29 +453,8 @@ func (r *TestResources) sendHealthRequest(base *url.URL, healthCheck func(resp *
 func (r *TestResources) updateAndWaitTillCryostatAvailable(cr *operatorv1beta2.Cryostat) (*operatorv1beta2.Cryostat, error) {
 	ctx := context.Background()
 
-	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		var err error
-		cr, err = r.Client.OperatorCRDs().Cryostats(cr.Namespace).Get(ctx, cr.Name)
-		if err != nil {
-			return fmt.Errorf("failed to get Cryostat CR \"%s\": %s", cr.Name, err.Error())
-		}
-
-		cr.Spec.StorageOptions = &operatorv1beta2.StorageConfiguration{
-			PVC: &operatorv1beta2.PersistentVolumeClaimConfig{
-				Spec: &corev1.PersistentVolumeClaimSpec{
-					StorageClassName: nil,
-					Resources: corev1.ResourceRequirements{
-						Requests: corev1.ResourceList{
-							corev1.ResourceStorage: resource.MustParse("1Gi"),
-						},
-					},
-				},
-			},
-		}
-
-		cr, err = r.Client.OperatorCRDs().Cryostats(cr.Namespace).Update(context.Background(), cr)
```

Comment:
If I'm understanding correctly, this is the part that was problematic? ie this `Update()` call can fail, which returns `nil, SomeError`. Since the `return err` now has a non-nil return value, the `retry.RetryOnConflict()` fires again, but the `cr` reference has now been set to `nil` and this second attempt fails for that reason?

Reply:
https://github.com/cryostatio/cryostat-operator/blob/f84caf0e9452ae570c9afdfae6bba368ccdf5f6a/internal/test/scorecard/clients.go#L138

https://github.com/cryostatio/cryostat-operator/blob/f84caf0e9452ae570c9afdfae6bba368ccdf5f6a/internal/test/scorecard/clients.go#L171-L181

Oh my, this is it. When there is an error, the empty CR is returned :D

Reply:
That's correct

Reply:
It was setting the `cr` to nil that exists outside the scope of this callback